akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRolePartitioningSpec.scala

```diff
@@ -0,0 +1,171 @@
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode2
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode3
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode4
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode5
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+
+abstract class ClusterShardingMinMembersPerRoleSpec
+    extends ClusterShardingRolePartitioningSpec(ClusterShardingMinMembersPerRoleConfiguredConfig)
+
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode1 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode2 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode3 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode4 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode5 extends ClusterShardingMinMembersPerRoleSpec
+
+abstract class ClusterShardingRolePartitioningSpec(config: ClusterShardingMinMembersPerRoleConfig)
+    extends MultiNodeSpec(config)
+    with MultiNodeClusterSpec
+    with ImplicitSender {
+
+  import config._
+
+  override def initialParticipants: Int = roles.size
+
+  private val fourthAddress = node(fourth).address
+  private val fifthAddress = node(fifth).address
+
+  "Cluster Sharding with roles" must {
+
+    "start the cluster, await convergence, init sharding on every node: 2 data types, 'akka.cluster.min-nr-of-members=2', partition shard location by 2 roles" in {
+      awaitClusterUp(first, second, third, fourth, fifth)
+      enterBarrier(s"${roles.size}-roles-up")
+
+      runOn(first, second, third, fourth, fifth) {
+        ClusterSharding(system).start(
+          typeName = E1.TypeKey,
+          entityProps = TestActors.echoActorProps,
+          // nodes 1,2,3: role R1, shard region E1, proxy region E2
+          settings = ClusterShardingSettings(system).withRole("R1"),
+          extractEntityId = E1.extractEntityId,
+          extractShardId = E1.extractShardId)
+
+        // when run on first, second and third (role R1) proxy region is started
+        ClusterSharding(system).start(
+          typeName = E2.TypeKey,
+          entityProps = TestActors.echoActorProps,
+          // nodes 4,5: role R2, shard region E2, proxy region E1
+          settings = ClusterShardingSettings(system).withRole("R2"),
+          extractEntityId = E2.extractEntityId,
+          extractShardId = E2.extractShardId)
+      }
+    }
+
+    // NOTE the following passes without the change for https://github.com/akka/akka/issues/28177
+    // so we just confirm the test case
```

Comment:
:+1: (though perhaps not necessary to leave this comment here, noting it in the issue/PR might be sufficient?